app/models/response.rb

```diff
@@ -27,6 +27,10 @@ class Response < ApplicationRecord
   before_save(:normalize_answers)
   before_create :generate_shortcode
 
+  # rails counter_cache increments on creation but does not decrement on deletion
+  # since we need the counter cache, we'll manually decrement on deletion
```

Comment:
Please mention that acts_as_paranoid is the reason for this, and provide a link to the open Github issue. That way if it gets resolved in future we might could take this out.

Reply:
https://github.com/ActsAsParanoid/acts_as_paranoid/issues/39